packages/blade/src/components/Input/BaseInput/AnimatedBaseInputWrapper.native.tsx

```diff
@@ -86,18 +94,70 @@ const _AnimatedBaseInputWrapper: React.ForwardRefRenderFunction<
     }),
   };
 
+  const baseInputState =
+    rest.currentInteraction === 'focus'
+      ? 'focused'
+      : rest.currentInteraction === 'hover'
+      ? 'hovered'
+      : rest.isDisabled
+      ? 'disabled'
+      : 'default';
```

Comment:
maybe create a `getBaseInputState` function and put these in if-else? difficult to read this 🥲

Reply:
Done